code-challenges.js

```diff
@@ -33,9 +93,28 @@ var middleLetters2 = "rhinoceros"
 
 // --------------------5) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
-var numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
-var numbersToAdd3 = []
-// Expected output: []
+// var numbersToAdd1 = [2, 4, 45, 9]
+// // Excpected output: [2, 6, 51, 60]
+// var numbersToAdd2 = [0, 7, -8, 12]
+// // Expected output: [0, 7, -1, 11]
+// var numbersToAdd3 = []
+// // Expected output: []
+
+
+//create function that addes given array
+//map through array
+
+// const arrAdd = (array) =>{
+//     let newArray = []
+//     for(let n=0; n<array.length; n++){
+//       return newArray.push(array[n] + (array[n-1]))
+// }
+//
+// }
+console.log(arrAdd(numbersToAdd1))
+
+  // const arrAdd = (array) =>{
+  // let newValues = array.map((curr, i, array) => {
+  // return array[i] += array[i-1] ? array[i-1] : 0
+  // })
+  // }
```

Comment:
Don't overthink this one. Your logic is right there. Just need an anchor variable to hold your iterating values.